core/commands/show_commit.py

```diff
@@ -28,3 +29,36 @@ def run(self, edit):
         commit_hash = self.view.settings().get("git_savvy.show_commit_view.commit")
         content = self.git("show","--no-color", commit_hash)
         self.view.run_command("gs_replace_view_text", {"text": content, "nuke_cursors": True})
+
+
+class GsShowFileAtCommitCommand(WindowCommand, GitCommand):
+
+    def run(self, commit_hash, filepath):
+        self.filepath = filepath
+        repo_path = self.repo_path
+        currentview = self.window.active_view()
+        lang = currentview.settings().get('syntax')
+        view = self.window.new_file()
+        view.set_syntax_file(lang)
+        view.settings().set("commit_hash", commit_hash)
+        view.settings().set("filepath", filepath)
+        view.settings().set("git_savvy.repo_path", repo_path)
+        view.settings().set("word_wrap", False)
+        view.settings().set("line_numbers", True)
+        view.set_name(SHOW_COMMIT_TITLE.format(commit_hash[:7]))
+        view.set_scratch(True)
+        view.run_command("gs_show_commit_file")   
+    
+
+class GsShowCommitFile(TextCommand, GitCommand):
```

Comment:
I don't think there is a reason for a second command here.  Just add a second, well-named, method to the `GsShowFileAtCommitCommand` class, and `sublime.set_timeout_async(self.method_name, 0)`.